Bottom: 12ec3a9ade29336ecbe86f9e36a6e1d82bec50bb
Top:    1cae12c27dbe280d4c5c99f381b8164748cfd787
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 20:28:53 +0100

Workaround crash where contour is passed as NULL to poly_PreContour






---

diff --git a/src/polygon.c b/src/polygon.c
index c697f3e..5c2d1f9 100644
--- a/src/polygon.c
+++ b/src/polygon.c
@@ -202,6 +202,16 @@ original_poly (PolygonType * p)
       }
   }
   END_LOOP;
+  if (contour == NULL)
+    {
+      printf ("How did that escape - did the loop iterate zero times??\n");
+      POLYGONPOINT_LOOP (p);
+        {
+          printf ("Hello\n");
+        }
+      END_LOOP;
+      return NULL;
+    }
   poly_PreContour (contour, TRUE);
   /* make sure it is a positive contour */
   if ((contour->Flags.orient) != PLF_DIR)
